azuread/resource_user.go

```diff
@@ -0,0 +1,188 @@
+package azuread
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/Azure/azure-sdk-for-go/services/graphrbac/1.6/graphrbac"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azuread/azuread/helpers/ar"
+	"github.com/terraform-providers/terraform-provider-azuread/azuread/helpers/p"
+)
+
+func resourceUser() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceUserCreate,
+		Read:   resourceUserRead,
+		Update: resourceUserUpdate,
+		Delete: resourceUserDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"user_principal_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"display_name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"mail_nickname": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"account_enabled": {
+				Type:     schema.TypeBool,
+				Required: true,
+			},
+
+			"password_profile": {
+				Type:     schema.TypeSet,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"password": {
+							Type:      schema.TypeString,
+							Required:  true,
+							Sensitive: true,
+						},
+						"force_password_change": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
```

Comment:
since it's only possible to configure a single password (profile) for a given user - I think it'd be worth moving these fields to the top-level (which means we can ignore setting them in the state / the Set/List requirement here) - WDYT?

Reply:
Thanks for the review.

That makes sense, I'm going to change it.